### PR TITLE
Loading model function made and refactored for more concise code

### PR DESCRIPTION
**Pull request by lsig**
# Loading BERT models
In this PR I made a models folder with a models.py where either the user can load icebert or convbert. 
This code is designed to be extensible so the only change needed to add a model should be adding a model 
to the model dictionary. 

# How to test
- Ruff linting with gha 
- No unit tests

### CODE REVIEW

**Comment by lsig** [comment]
I think the Codacy SCA does not review the code after the lint action is done, which is a problem since this should not fail since black should already have formatted the code and removed trailing whitespaces. But it seems that the lint.yml is fixed since I intentionally had a line to long to see if the lint check would fail but it did not.

File: models/models.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    AutoTokenizer,
+    BertModel,
+    ConvBertForQuestionAnswering,
+    RobertaForQuestionAnswering,
+)
+
+from training.utils import get_device
+
+MODELS = {
+    "icebert":  { "name": "Mideind/icebert", "model_type": RobertaForQuestionAnswering},
```
Comment (njallskarp) [inline models/models.py]: replace model_type with bert_model

File: models/models.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "convbert": { "name": "jonfd/convbert-base-igc-is", "model_type": ConvBertForQuestionAnswering}
+}
+
+def load(model_name: str) -> tuple[BertModel, Encoding]:
+    if model_name not in MODELS: 
+        raise ValueError(f"Invalid model name: {model_name}. Supported models: {list(MODELS.keys())}")
+
+    BERT_MODEL = MODELS[model_name]
```
Comment (njallskarp) [inline models/models.py]: these three are not constants, use regular snake case instead

**Review by njallskarp — CHANGES_REQUESTED** [review]
noice - two added comments for changes

File: models/models.py (hunk 1 above)
Comment (lsig) [reply models/models.py]: See latest commit

File: models/models.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+from training.utils import get_device
+
+MODELS = {
+    "icebert":  { "name": "Mideind/icebert", "bert_type": RobertaForQuestionAnswering},
+    "convbert": { "name": "jonfd/convbert-base-igc-is", "bert_type": ConvBertForQuestionAnswering}
+}
+
+def load(model_name: str) -> tuple[BertModel, Encoding]:
+    if model_name not in MODELS:
```
Comment (Eysteinn-Orn) [inline models/models.py]: seems to be a trailing whitespace that is not getting fixed by the linter in this line.

Comment (lsig) [reply models/models.py]: See comment above

Comment (Eysteinn-Orn) [reply models/models.py]: yes sorry mb!

**Review by njallskarp — APPROVED** [review]

**Pull request merged**